src/Core/Managed/Shared/Extensibility/Metrics/DependencyMetricExtractor.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using System.Collections.Generic;
+
+namespace Microsoft.ApplicationInsights.Extensibility.Metrics
+{
+    public class DependencyMetricExtractor : MetricExtractorTelemetryProcessorBase
```

Comment:
The other way to design this would be to have single metric processor called "MetricExtractorTelemetryProcessor" and have a list of metric extractors within that. Each approach has certain pros and cons. I'd like you to consider those and make a decision (whatever that would be). If you already considered and made a choice - I'd be interested to seewhy this coded approach won.